src/Features/CSharp/Portable/EditAndContinue/CSharpEditAndContinueAnalyzer.cs

```diff
@@ -539,6 +541,7 @@ protected override Match<SyntaxNode> ComputeBodyMatch(SyntaxNode oldBody, Syntax
 
                 static SyntaxNode GetMatchingRoot(SyntaxNode body)
                 {
+                    RoslynDebug.Assert(body.Parent is object);
```

Comment:
>RoslynDebug.Assert(body.Parent is object); [](start = 20, length = 42)

I think it'd be better to consistently use Parent! rather than adding asserts in some places and ! in others.